src/branch_and_bound.rs

```diff
@@ -7,14 +7,28 @@ use std::convert::TryInto;
 pub struct OutPointValue(pub OutPoint, pub u64);
 
 pub struct BranchAndBound {
+    /// The spending target to be reached. Note that this variable should hold only the sum of the
+    /// outputs, and not the fees: those will be calculated using `addresses_num`,
+    /// `estimated_fee_rate`, `size_of_header`, `size_per_output`, `size_per_input`.
     pub spending_target: u64,
+    /// Utxos in this vector will be always included in the result.
+    /// Useful for RBF.
     pub mandatory_utxos: Vec<OutPointValue>,
+    /// Utxos that could be included in the result, if mandatory_utxos
+    /// are not enough.
     pub optional_utxos: Vec<OutPointValue>,
+    /// How many output the transaction has. Used for estimating fees.
     pub addressees_num: u64,
-    pub estimated_fees: u64,
+    /// Estimated fee rate, in sat/vbyte
+    pub estimated_fee_rate: u64,
+    /// Size of the transaction header. Used for estimating fees.
     pub size_of_header: u64,
+    /// Size of each transaction output. Used for estimating fees.
     pub size_per_output: u64,
```

Comment:
Inputs can be assumed to be the same size, but for outputs that is a dangerous assumption. E.g. if you are sending to a multisig address and returning change to your own wallet, the two outputs will have differing sizes.